src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -0,0 +1,928 @@
+class X
+{
+    static Dictionary<Dictionary<int, int>, List<int>> v1 = new()
+    {
+        { new() { [1] = 2 }, new() { 3, 4 } }
+    };
+
+    static void Print(Dictionary<int, int> v)
+    {
+        foreach (var pair in v)
+            Console.Write($""{pair.Key}{pair.Value}"");
+    }
+
+    static void Print(List<int> v)
+    {
+        foreach (var item in v)
+            Console.Write(item);
+    }
+
+    public static void Main()
+    {
+        foreach (var pair in v1)
+        {
+            Print(pair.Key);
+            Print(pair.Value);
+        }
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics();
+
+            CompileAndVerify(comp, expectedOutput: "1234");
+        }
+
+        [Fact]
+        public void TestImplicitConversion()
+        {
+            var comp = CreateCompilation(@"
+class Dog
+{
+    public Dog() {}
+}
+class Animal
+{
+    private Animal() {}
+    public static implicit operator Animal(Dog dog) => throw null;
+}
+
+class Program
+{
+    public static void M(Animal a) => throw null;
+    public static void Main()
+    {
+        M(new());
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (17,11): error CS0122: 'Animal.Animal()' is inaccessible due to its protection level
+                //         M(new());
+                Diagnostic(ErrorCode.ERR_BadAccess, "new()").WithArguments("Animal.Animal()").WithLocation(17, 11)
+                );
+        }
+
+        [Fact]
+        public void TestOverloadResolution()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    public C(int i) {}
+}
+
+class D
+{
+}
+
+class Program
+{
+    static void M(C c, object o) {}
+    static void M(D d, int i) {}
+
+    public static void Main()
+    {
+        M(new(1), 1);
+    }
+}
+", options: TestOptions.ReleaseExe).VerifyDiagnostics(
+                // (18,11): error CS1729: 'D' does not contain a constructor that takes 1 arguments
+                //         M(new(1), 1);
+                Diagnostic(ErrorCode.ERR_BadCtorArgCount, "new(1)").WithArguments("D", "1").WithLocation(18, 11)
+                );
+        }
+
+        [Fact]
+        public void TestNullableType1()
+        {
+            var comp = CreateCompilation(@"
+using System;
+struct S
+{
+    public S(int i)
+    {
+        Console.Write(i);
+    }
+
+    public static void Main()
+    {
+        S? s = new(43);
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "43");
+        }
+
+        [Fact]
+        public void TestNullableType2()
+        {
+            var comp = CreateCompilation(@"
+using System;
+struct S
+{
+    public static T? M<T>() where T : struct
+    {
+        return new();
+    }
+
+    public static void Main()
+    {
+        Console.Write(M<S>());
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "S");
+        }
+    }
+}
```

Comment:
Aside from adding a few more tests (which I'm listing in [https://github.com/dotnet/roslyn/issues/28489](https://github.com/dotnet/roslyn/issues/28489)), the main thing I would add here is some verification of the semantic model. `GetDeclaredType` (null), `GetTypeInfo` (no natural type, but a proper converted type, except for some error cases). 
This doesn't need to be added to every test, just a well-selected few. I'm ok to keep this for a later PR (assuming other reviewers ok too).